tests/userLogin.test.ts

```diff
@@ -0,0 +1,36 @@
+import request from "supertest";
+
+import { Role } from "../src/models/user.model";
+
+import { STATUS_CODES } from "../src/utils/constants";
+
+import { PrismaUser, prismaMock } from "../singleton";
+import app from "../src/server";
+
+import ResolvedValue = jest.ResolvedValue;
+
+describe("User login", () => {
+  const email = "test1@test.com";
+  const password = "Test123Pass!";
+
+  it("Should log in an existing user", async () => {
+    const email = "test1@test.com";
+    const password = "Test123Pass!";
+
+    const data = { email, password };
+    const existingUser = { email, password } as ResolvedValue<PrismaUser>;
+
+    prismaMock.user.findUnique.mockResolvedValue(existingUser);
+
+    await request(app).post("/api/login").send(data).expect(STATUS_CODES.OK);
+  });
+
+  it("Should not log in non existing user", async () => {
+    const data = { email: "test2@test.com", password: "Test123456Pass?" };
```

Comment:
You can use those declared variables (email and password) in the upper scope.

Reply:
I thought I had to use different email. Fixed. 

I'm still not sure how this test works (`Should not log in non existing user`). I guess because there is no mocked resolved value, it is behaving as there is nothing returned from the database, which means this user doesn't exist in the database. Hence, we get `unauthorized`.

Is it something like this?

Reply:
For this case you have to write two tests. One is when there is no user in database (here you can mock prisma to return nothing like in those registration endpoint tests). Second test is when user is in database but password is not correct. You should mock either `isValidPassword` function to return you falsy value.